spec/models/concerns/digital_object/xml_datastream_rendering_spec.rb

```diff
@@ -0,0 +1,363 @@
+        '<?xml version="1.0"?>
+        <mods:mods>
+          <mods:name ID="jds1329 Salinger, J. D." valueURI="http://id.loc.gov/authorities/names/n50016589">
+            <mods:namePart>Salinger, J. D.</mods:namePart>
+          </mods:name>
+          <mods:name ID="Lincoln, Abraham" valueURI="http://id.loc.gov/authorities/names/n79006779">
+            <mods:namePart>Lincoln, Abraham</mods:namePart>
+         </mods:name>
+        </mods:mods>'
+      end
+
+      it 'generates corrext xml' do
+        digital_object.render_xml_translation_with_data(doc, doc, base_xml_translation, dynamic_field_data, xml_translation_map)
+        expect(doc).to be_equivalent_to expected_mods
+      end
+    end
+  end
+
+  describe '#value_with_substitutions' do
+    let(:name_df_data) { dynamic_field_data["name"][0] }
+
+    it "replaces value when its the only thing in the string" do
+      expect(digital_object.value_with_substitutions("{{name_term.value}}", name_df_data)).to eql "Salinger, J. D."
+    end
+
+    it "replaces correct value when there are multiple references" do
+      expect(digital_object.value_with_substitutions("{{name_term.value}}{{name_term.uni}}", name_df_data)).to eql "Salinger, J. D.jds1329"
+    end
+
+    it "replaces corrext value when reference is surrounded by characters" do
+      expect(digital_object.value_with_substitutions("Author name is {{name_term.value}}.", name_df_data)).to eql "Author name is Salinger, J. D.."
+    end
+  end
+
+  describe "#value_for_field_name" do
+    let(:name_df_data) { dynamic_field_data["name"][0] }
+    let(:title_df_data) { dynamic_field_data["title"][0] }
+
+    before do
+      allow(digital_object).to receive(:project).and_return(double(display_label: "Test Project"))
+    end
+
+    it "looks up value when it starts with a $" do
+      expect(digital_object.value_for_field_name("$project.display_label", name_df_data)).to eql "Test Project"
+    end
+
+    it "returns data unavailable when referenced value doesn't have a matching variable" do
+      expect(digital_object.value_for_field_name("$project.expiration_date", name_df_data)).to eql "Data unavailable"
+    end
+
+    it "returns correct value when fields are nested" do
+      expect(digital_object.value_for_field_name("name_term.uni", name_df_data)).to eql "jds1329"
+    end
+
+    it "returns correct value when not nested" do
+      expect(digital_object.value_for_field_name("title_sort_portion", title_df_data)).to eql "Catcher in the Rye"
+    end
+
+    it "returns empty string if field does not exist" do
+      expect(digital_object.value_for_field_name("name_term.last_name", name_df_data)).to eql ""
+    end
+  end
+
+  describe "#render_output_of_ternary" do
+    let(:name_df_data) { dynamic_field_data["name"][0] }
+
+    it 'returns "true" value if field present' do
+      arr = ["name_term.value", "True", "False"]
+      expect(digital_object.render_output_of_ternary(arr, name_df_data)).to eql "True"
```

Comment:
Would you mind changing the "True" and "False" string values in the render_output_of_ternary tests to "Yes" and "No" (or something else), just so when looking at these tests, there's no chance that someone would incorrectly assume that render_output_of_ternary actually returns boolean values as strings?  I think it might make the functionality clearer at first glance.

Reply:
Will do